### PR TITLE
NextJS 13.3 app directory & frameworks CTA

### LINKED ISSUE #5660: Why does firebase deploy list a static page as non-static
### [REQUIRED] Environment info

When I try to deploy a next project, the pages marked as static by next build are given as reasons why cloud functions are needed (subsequently causing cold start delays rather than serving static pages from cdn).

**firebase-tools:** using 11.19.0 but tried others as well

**Platform:** Windows

### [REQUIRED] Steps to reproduce

First I'm hoping someone can just clarify if I am misunderstanding something (see the actual behavior below).

### [REQUIRED] Expected behavior

I expect cloud functions to only apply to the api routes and the server side rendered pages.

### [REQUIRED] Actual behavior

I have a Next.js 12 project, with a page that has minimal stuff in it:
```
const Dashboards = () => {
  return <div>This is the Dashboards page</div>;
};

export default Dashboards;
```

but then firebase deploy lists this as a reason why it needs cloud functions even though next.js lists this as a static page.

```
Route (pages)                              Size     First Load JS
┌   /_app                                  0 B             349 kB
├ λ /[provider]                            5.61 kB         355 kB
├ ○ /404                                   194 B           350 kB
├ λ /api/hello                             0 B             349 kB
├ ○ /charts (657 ms)                       73.3 kB         423 kB
├ ○ /dashboards                            274 B           350 kB
├ ○ /our-approach                          1.15 kB         351 kB
├ ○ /research                              344 B           350 kB
├ ○ /search (1804 ms)                      272 B           350 kB
├ ○ /security                              1.37 kB         351 kB
└ ○ /user/settings                         4.99 kB         354 kB
+ First Load JS shared by all              350 kB
  ├ chunks/framework-4556c45dd113b893.js   45.2 kB
  ├ chunks/main-d2fa5d94e1c1d6ef.js        31.3 kB
  ├ chunks/pages/_app-2bbab969a0265f20.js  271 kB
  ├ chunks/webpack-7366c3a031b580cf.js     1.88 kB
  └ css/d25682aacd1f4b07.css               1.09 kB

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)

Building a Cloud Function to run this application. This is needed due to:
 • non-static route /api/hello
 • non-static route /charts
 • non-static route /dashboards
 • non-static route /our-approach
 • non-static route /research
 • and 5 other reasons, use --debug to see more
```
I deleted the next.config.js file, tried different versions of firebase, downgraded next from 13 to 12 and eliminated other elements of next that I though might be contributing (like font optimization). I think there is something basic about firebase webframework deploy that I am not understanding.

### PR DESCRIPTION
**Pull request by jamesdaniels**
Next 13.3 app directory changed in a couple important ways:
* Can have prerendered content
* Can have static files
* Can have API routes

This broke our integration, see #5661 

While I'm mucking around in here:
* There were some issues with reasons for backend, fixes #5660
* Beautify the Frameworks detection banner, include links to docs etc.

### CODE REVIEW

File: src/frameworks/next/index.ts [hunk 1]
```diff
@@ -175,6 +178,27 @@ export async function build(dir: string): Promise<BuildResult> {
...
+    join(dir, distDir, APP_PATH_ROUTES_MANIFEST)
+  ).catch(() => ({}));
+  await Promise.all(
+    Object.entries(appPathRoutesManifest).map(async ([key, source]) => {
+      if (basename(key) !== "route") return;
+      const parts = source.split("/").filter((it) => !!it);
+      const partsOrIndex = parts.length > 0 ? parts : ["index"];
+      const routePath = join(dir, distDir, "server", "app", ...partsOrIndex);
+      const metadataPath = `${routePath}.meta`;
+      if (
```
Comment (jamesdaniels) [inline src/frameworks/next/index.ts:191]: move to sync, it's already in a promise

File: src/frameworks/next/index.ts [hunk 2]
```diff
@@ -312,35 +342,48 @@ export async function ɵcodegenPublicDirectory(sourceDir: string, destDir: strin
...
+      let sourcePath = join(contentDist, ...partsOrIndex);
+      let destPath = join(destDir, ...partsOrIndex);
+      if (!fileExistsSync(sourcePath) && fileExistsSync(`${sourcePath}.html`)) {
+        sourcePath += ".html";
+        destPath += ".html";
+      } else if (appPathRoute && basename(appPathRoute) === "route" && dirExistsSync(sourcePath)) {
+        sourcePath += ".body";
+      }
+
+      if (!pathExistsSync(sourcePath)) {
```
Comment (jamesdaniels) [inline src/frameworks/next/index.ts:370]: pull into an else?

File: src/frameworks/next/utils.ts [hunk 3]
```diff
@@ -193,15 +194,17 @@ export async function isUsingMiddleware(dir: string, isDevMode: boolean): Promis
  * @param dir path to `distDir` - where the manifests are located
  */
 export async function isUsingImageOptimization(dir: string): Promise<boolean> {
-  const { isNextImageImported } = await readJSON<ExportMarker>(join(dir, EXPORT_MARKER));
+  let { isNextImageImported } = await readJSON<ExportMarker>(join(dir, EXPORT_MARKER));
+  // App directory doesn't use the export marker, look it up manually
+  if (!isNextImageImported && isUsingAppDirectory(dir)) {
+    isNextImageImported = (await readFile(join(dir, "server", "client-reference-manifest.js")))
+      .toString()
+      .includes("node_modules/next/dist/client/image.js");
```
Comment (jamesdaniels) [inline src/frameworks/next/utils.ts:202]: TODO be smarter about this, should we read the json, require.resolve the client name?

**Review by leoortizz — APPROVED** [review]
:shipit:

**Pull request merged**